Add gamma curves settings and explicit refresh button

BrunoWallner opened the pull request:
I added the feature of gamma curves, which basically just change the curve of the brightness of the slider that gets sent to the monitor. With a gamma curve of 2.0, the brightness sent to the monitor gets powered with 2: `new brightness = brightness.powf(2.0)`.
This makes changes at the lower end of the slider slower, which is desired on my kind of display.
This gets saved to `~/.config/cosmic/` using `libcosmic::cosmic_config`.
I also restructured the way that the displays gets fetched. The logic should have stayed the same.

I also added an explicit button to refetch the displays and reload the config, in case a Monitor has been plugged in.
![Screenshot_2025-06-23_19-51-53](https://github.com/user-attachments/assets/eb6f8ef3-1c6f-449d-a053-c9cd74f7dc88)

wiiznokes commented:
Hi, thanks for the PR, i think this could be an useful feature for some displays.
How ever, i have some remarks

If the gamma curve is only relevant in some display, and is a thing that you configure one time, i don't think it make sens to display it unconditionally on the applet. i think we can take inspiration from https://github.com/emoacht/Monitorian, with a settings popup when you right click the applet. Also, since it a per display config, i think we should group it in the ui. Probably like this:
```
icon slider x %
   gamma_curve_title slider x %
---------------------
Dark theme switch
```
Maybe we could add a name like Monitorian?


Also, i think the refresh button should live in the setting pop up.
- The refresh button should only refresh display, not the config. We can use a subscription for that i think. i'm not even sure we should re set the monitor brightness after a `Message::UpdateMonitors`.
- The refresh button cause a lot of ` WARN Error sending event to subscription: TrySendError { kind: Full } `, you're probably calling some blocking code on the async runtime.
- if think the `brightness` field should stay u16 between 0..100
- gamma_curve should not be stored in the Monitor struct, do this instead, and only use this as the source of truth for gamma_curve

```rs
#[derive(Clone, CosmicConfigEntry, Debug, Deserialize, PartialEq, Serialize)]
#[serde(default)]
pub struct Config {
    pub monitors: Vec<Monitor>,
}

#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
pub struct Monitor {
    pub id: String,
    pub gamma_curve: f32,
}
```

BrunoWallner commented:
Now all this is changed:
- The config gets watched and gets updated automatically
- Changed gamma curve to gamma map
- Gamma map is now grouped to the brightness slider, but hidden by default
  - The brightness icon now supports mouse wheel input, left click toggles minmax brightness and right click expands the monitor settings (currently only the gamma map slider)
- Removed the refresh button entirely. I'm not sure where to put it
- The position of the monitor name tooltip is now right
- The gamma map value for each monitor is stored in the config

It currently looks like this:
![Screenshot_2025-07-03_10-49-50](https://github.com/user-attachments/assets/be9ed129-d106-4b2f-b8c1-02696bf01bce)

I currently still use f32 for brightness on frontend Monitor.
There is still some junk in the code due to unused refresh logic.

wiiznokes commented:
Hey, i've just merged https://github.com/cosmic-utils/cosmic-ext-applet-external-monitor-brightness/pull/23, which is based on your branch with some improvements
To name a few
- there is now a clear separation between ui state, config and subscription state.
- refresh is implemented, in a separate popup
- the ui doesn't use variable declaration. I think it's easier to read like that.

let me know what you think
